site/gatsby-site/src/components/discover/Hits.js

```diff
@@ -1,21 +1,32 @@
-import React, { useEffect, useState } from 'react';
-import { connectHits, connectStateResults } from 'react-instantsearch-dom';
+import React, { useMemo } from 'react';
+import { useHits, useInstantSearch, useRefinementList } from 'react-instantsearch';
 import Hit from './Hit';
 import { DisplayModeEnumParam } from './queryParams';
 import { useQueryParam } from 'use-query-params';
 import CardSkeleton from 'elements/Skeletons/Card';
 import ListSkeleton from 'elements/Skeletons/List';
+import { VIEW_TYPES } from 'utils/discover';
+
+export default function Hits({ ...props }) {
+  const { status, results, indexUiState } = useInstantSearch();
+
+  const { hits } = useHits(props);
 
-const Hits = ({ hits, isSearchStalled, toggleFilterByIncidentId, viewType }) => {
   const [display] = useQueryParam('display', DisplayModeEnumParam);
 
-  const [isMounted, setIsMounted] = useState(false);
+  const { refine } = useRefinementList({ attribute: 'incident_id' });
+
+  const isLoading = status === 'loading' || status === 'stalled';
 
-  useEffect(() => {
-    setIsMounted(true);
-  }, []);
+  const viewType = useMemo(() => {
+    return indexUiState.configure.distinct === true &&
+      indexUiState.refinementList.is_incident_report.length > 0 &&
+      indexUiState.refinementList.is_incident_report[0] === 'true'
+      ? VIEW_TYPES.INCIDENTS
+      : VIEW_TYPES.REPORTS;
+  }, [indexUiState]);
 
-  if (!isSearchStalled && hits.length === 0) {
+  if (!results.__isArtificial && results.nbHits === 0) {
```

Comment:
sorry, I don't understand results.__isArtificial, and can't find it in documentation:
https://instantsearchjs.netlify.app/algoliasearch-helper-js/reference.html#searchresults
https://www.algolia.com/doc/api-reference/widgets/use-instantsearch/react/

Pointer or explanation please?

Reply:
It is buried in their docs, and the prop name is so weird:
https://www.algolia.com/doc/ui-libraries/autocomplete/guides/implementing-multiple-search-states/#implementing-the-no-results-search-state